tests/framework/extensions/provisioning/verify.go

```diff
@@ -37,6 +40,8 @@ import (
 const (
 	logMessageKubernetesVersion = "Validating the current version is the upgraded one"
 	hostnameLimit               = 63
+	machineNameAnnotation       = "cluster.x-k8s.io/machine"
+	publicIPAnnotation          = "rke2.io/external-ip"
```

Comment:
This annotation is likely `k3s.io/external-ip` for k3s clusters: https://github.com/k3s-io/k3s/blob/550dd0578f79882e1a78d8468fdbefa95faa145c/pkg/cloudprovider/instances.go#L17